### ISSUE TITLE
No visible @interface for 'MobileRTC' 

### ISSUE THREAD

**Issue by Palwisha77** [chaimPaneth/react-native-zoom-bridge#20]
I have recreated the project and run the command "pod install" then remove incorrect headers from all SDK files after that I build the project and getting the error "No visible @interface for 'MobileRTC' declares the selector 'setMobileRTCDomain:'".
Please help me in this I am trying to integrate IOS SDK for the last 3 to 4 days but still unable to integrate.

**Comment by chaimPaneth** [comment]
Hi @Palwisha77 the issue is a breaking change in the SDK, are you using the scripts provided by this library to import the SDK or manually downloading them?

**Comment by Palwisha77** [comment]
I have downloaded the SDK manually from the link provided in the library. The following are the versions that I am using in my project.
SDK version is v5.0.24433.0616
React Native Version is 0.61.5
Xcode Version is 11.3.1

**Comment by chaimPaneth** [comment]
Try with this version v4.6.21666.0428 download it from here

ios-dev: https://github.com/zoom/zoom-sdk-ios/archive/v4.6.21666.0428.zip
ios-prod: https://github.com/zoom/zoom-sdk-ios/archive/v4.6.21666.0428.zip
android: https://github.com/zoom/zoom-sdk-android/archive/v4.6.21666.0429.zip

put them in `node_modules/react-native-zoom-bridge/ios/libs` && `node_modules/react-native-zoom-bridge/android/libs`
then run `pod install` then rename the headers for ios.

Tell me if it worked.

**Comment by chaimPaneth** [comment]
@Palwisha77 Iv'e pushed a fix to the library version 1.0.17 and changed the SDK import bash files to download previous working version v4.6.21666.0428.

I will update the library to work with the latest SDK when I get the chance.

**Comment by Palwisha77** [comment]
Thank you I will try this and let you know if it works.

**Comment by Palwisha77** [comment]
@chaimPaneth  I used the library version 1.0.17 and SDK version v4.6.21666.0428 the build is successful but the app won't install in device. The provisioning profile and signing certificates are correct as the app was installed successfully without the SDK but after integrating the SDK app won't install. Is there any signing issue in SDK or Frameworks? I don't understand the issue.

**Comment by chaimPaneth** [comment]
@Palwisha77 when installing on a real device you have to use the prod SDK and when running in the simulator you have to use the dev SDK thats why there are two bash scripts for iOS make sure you use the correct one.

1: Import (correct) SDK using bash file
2: Run pod install
3: Open xcode and find & replace "MobileRTC/" with ""
4: Rerun

**Comment by Palwisha77** [comment]
Thank you for this I have used dev  SDK and trying to install it on the device.

**Comment by chaimPaneth** [comment]
Tell me if it works with the prod SDK

**Comment by Palwisha77** [comment]
unable to install the app on device with prod SDK also. here is the error details.
![Screenshot 2020-07-14 at 10 11 21 AM](https://user-images.githubusercontent.com/25461981/87386994-395a5900-c5bb-11ea-8956-3bd0ace97aa6.png)
IOS device: iPhone 6
version 12.4.5

**Comment by chaimPaneth** [comment]
@Palwisha77 the Failure Reason mentions "No code signature found." there is probably an issue with your code signing, when installing to a device from xcode you need a development certificate not adhoc or production also setting it on auto signing might help.

**Comment by Palwisha77** [comment]
@chaimPaneth I think my certificates are correct because the app is installing on the IOS device successfully without the SDK but after integrating I got this issue.

**Comment by chaimPaneth** [comment]
Can you try removing the library `npm uninstall react-native-zoom-bridge` then rerun `pod install` and see if it will run on a real device?

**Comment by Palwisha77** [comment]
@chaimPaneth as u said I unliked the library and run the app on a real device. App installed successfully.

**Comment by chaimPaneth** [comment]
@Palwisha77 did you install the production SDK before running the app on a real device, check this out https://github.com/chaimPaneth/react-native-zoom-bridge#sdk-type.

```
$ npm i react-native-zoom-bridge --save
$ chmod +x ./node_modules/react-native-zoom-bridge/bin/import_prod_sdk.sh
$ ./node_modules/react-native-zoom-bridge/bin/import_prod_sdk.sh
$ cd ios
$ pod install
```

Rerun project.

**Comment by Palwisha77** [comment]
Yes, I have tried both dev SDK and prod SDK and got the same issue with both SDK types.

**Comment by chaimPaneth** [comment]
@Palwisha77 did you check this out https://stackoverflow.com/questions/38371044/ios-app-testing-app-installation-failed-no-code-signature-found or this https://developer.apple.com/forums/thread/60563

**Comment by Palwisha77** [comment]
I have checked this https://stackoverflow.com/questions/38371044/ios-app-testing-app-installation-failed-no-code-signature-found  but my Xcode version is  11.3.1 (11C504). I didn't find any 'Embed Frameworks' under 'Build Phase'
![Screenshot 2020-07-20 at 1 36 37 PM](https://user-images.githubusercontent.com/25461981/87917220-0e29ab00-ca8e-11ea-823f-a192f3042ff4.png)
Instead of you can see there is 'Embed Pods Framework' which includes the framework path.
![Screenshot 2020-07-20 at 1 35 24 PM](https://user-images.githubusercontent.com/25461981/87917628-b63f7400-ca8e-11ea-85af-108594d4e52b.png)

**Comment by chaimPaneth** [comment]
Hi @Palwisha77 

Can you try and archive an .ipa with AdHoc provision and upload it to Microsoft AppCenter and install from there?

Tell me if archiving worked?

**Comment by Palwisha77** [comment]
No, I have not try archiving but soon will tell you.

**Comment by jordanwade** [comment]
Just wanted to note I'm having the same issue.

**Comment by thanh125643** [comment]
me too but you can run it with archiving and install from Microsoft AppCenter

**Comment by jordanwade** [comment]
I was able to get it to work in dev on a locally tethered device by removing the reference to `MobileRTC.framework` in the Input and Output paths under `[CP] Embed Pods Frameworks` in `Build Phases`. I'm no Xcode expert, so I don't know if that needs to be there or not, but that is what worked for me currently for testing on a physical device.

**Comment by dearsikandarkhan** [comment]
> I was able to get it to work in dev on a locally tethered device by removing the reference to `MobileRTC.framework` in the Input and Output paths under `[CP] Embed Pods Frameworks` in `Build Phases`. I'm no Xcode expert, so I don't know if that needs to be there or not, but that is what worked for me currently for testing on a physical device.

This worked for me 💯

**Comment by dearsikandarkhan** [comment]
This issue occured again for me recent build. I fixed it by following steps 
- Remove /DerivedData 
- pod deintegrate 
- follow https://github.com/chaimPaneth/react-native-zoom-bridge/issues/20#issuecomment-670713511 

👍

**Comment by socheat-leang** [comment]
> I was able to get it to work in dev on a locally tethered device by removing the reference to `MobileRTC.framework` in the Input and Output paths under `[CP] Embed Pods Frameworks` in `Build Phases`. I'm no Xcode expert, so I don't know if that needs to be there or not, but that is what worked for me currently for testing on a physical device.

It's not work for me. Anyone have other solution beside this. ?
Build success,can install to simulator. but got error when install into physical device. 


`Details

Unable to install "testingapp"
Domain: com.apple.dt.MobileDeviceErrorDomain
Code: -402620388
--
No code signature found.
Domain: com.apple.dt.MobileDeviceErrorDomain
Code: -402620388
User Info: {
    DVTRadarComponentKey = 261622;
    MobileDeviceErrorCode = "(0xE800801C)";
    "com.apple.dtdevicekit.stacktrace" = (
	0   DTDeviceKitBase                     0x000000012a667c8f DTDKCreateNSErrorFromAMDErrorCode + 220
	1   DTDeviceKitBase                     0x000000012a6a6241 __90-[DTDKMobileDeviceToken installApplicationBundleAtPath:withOptions:andError:withCallback:]_block_invoke + 155
	2   DVTFoundation                       0x000000010ebb564b DVTInvokeWithStrongOwnership + 71
	3   DTDeviceKitBase                     0x000000012a6a5f82 -[DTDKMobileDeviceToken installApplicationBundleAtPath:withOptions:andError:withCallback:] + 1440
	4   IDEiOSSupportCore                   0x000000012a516a10 __118-[DVTiOSDevice(DVTiPhoneApplicationInstallation) processAppInstallSet:appUninstallSet:installOptions:completionBlock:]_block_invoke.292 + 3513
	5   DVTFoundation                       0x000000010ece417e __DVT_CALLING_CLIENT_BLOCK__ + 7
	6   DVTFoundation                       0x000000010ece5da0 __DVTDispatchAsync_block_invoke + 1191
	7   libdispatch.dylib                   0x00007fff6efb56c4 _dispatch_call_block_and_release + 12
	8   libdispatch.dylib                   0x00007fff6efb6658 _dispatch_client_callout + 8
	9   libdispatch.dylib                   0x00007fff6efbbc44 _dispatch_lane_serial_drain + 597
	10  libdispatch.dylib                   0x00007fff6efbc5d6 _dispatch_lane_invoke + 363
	11  libdispatch.dylib                   0x00007fff6efc5c09 _dispatch_workloop_worker_thread + 596
	12  libsystem_pthread.dylib             0x00007fff6f214a3d _pthread_wqthread + 290
	13  libsystem_pthread.dylib             0x00007fff6f213b77 start_wqthread + 15
);
}
--`
